### PR TITLE
refactor(simapp): re-wire autocli with latest client/v2 changes

### PR DESCRIPTION
**Pull request by julienrbrt**
## Description

We have just closed AutoCLI txs support and are migrating our modules to use it.
For having the best AutoCLI experience (ie automatic key resolving of input) and supporting txs, there is some extra wiring to add in the root.go

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: scripts/go-mod-tidy-all.sh
Comment (julienrbrt) [inline scripts/go-mod-tidy-all.sh:1]: I am used to call this in the SDK as well instead of going manually to each sub-modules

**Review by damiannolan — APPROVED** [review]
Thanks @julienrbrt

**Review by colin-axner — APPROVED** [review]
Thanks @julienrbrt! :heart:

**Comment by julienrbrt** [comment]
Oh I forgot this 😅 . I'll fix the conflicts

**Pull request merged**